server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -875,16 +875,14 @@ protected Node(
             );
             TopNSearchTasksLogger taskConsumer = new TopNSearchTasksLogger(settings, settingsModule.getClusterSettings());
             transportService.getTaskManager().registerTaskResourceConsumer(taskConsumer);
-            if (FeatureFlags.isEnabled(FeatureFlags.EXTENSIONS)) {
```

Comment:
Why are we removing the feature flag from here?

Reply:
The flag is redundant with the NoopExtensionManager

Reply:
Not sure why `NoopExtensionManager` was added in the first place. Can we remove `NoopExtensionManager` completely and add the flag back? WDYT?

Reply:
Feature flags are structural artifacts that make OpenSearch harder test and troubleshot by introducing duplicate code paths - while needed, the less branching codepaths the easier to maintain.

By depending on feature flags classes such as Node or IndicesService are responsible for checking is extensions enabled, when for the most part the answer is those classes shouldn't need to know if that is the case (there is the clear exception when the ExtensionManager is initialized).

Noop implementation invert the responsibility of the feature flag, so only the single implementing class needs to know rather than the callers of this class.

I am strongly in the noop implementation camp.

Reply:
I'm okay with removing the feature flag.  When I was adding all the checks, I erred on the side of caution and had them for every time that extensions interacted with code outside the directory.